[Kaka0Adult] 장바구니 추가, 삭제 기능 구현

hyo-jin-jeong opened the pull request:
>wecode PR message template 및 체크 리스트 입니다. 
아래 사항들을 전부 작성/체크 하시고 PR 해주세요!

## 수정 사항 간략한 한줄 요약
(여기에 수정 사항에 대한 간략한 한줄 요약/제목 작성해 주세요.)
- 장바구니 추가, 삭제 기능 구현
		 
## 수정 사항들 자세한 내용
(여기에 수정 사항에 대한 자세한 내용을 작성해 주세요.)
- BasketView 생성
- 장바구니 추가 post method 생성
<img width="373" alt="스크린샷 2021-06-11 오후 2 03 05" src="https://user-images.githubusercontent.com/55984573/121633463-ce145200-cabd-11eb-9c33-be5226da4799.png">
<img width="261" alt="스크린샷 2021-06-11 오후 2 03 12" src="https://user-images.githubusercontent.com/55984573/121633477-d66c8d00-cabd-11eb-9fa0-2f77242a8685.png">

- 장바구니 삭제 delete method 생성
<img width="375" alt="스크린샷 2021-06-11 오후 2 07 44" src="https://user-images.githubusercontent.com/55984573/121633773-66aad200-cabe-11eb-9ed4-149e8c9ae783.png">
<img width="248" alt="스크린샷 2021-06-11 오후 2 08 51" src="https://user-images.githubusercontent.com/55984573/121633859-8e9a3580-cabe-11eb-83c9-d3ae754ada5b.png">

- 예외처리 


## 기타 질문 및 특이 사항
(궁금한 사항들, 하면서 느낀 점들 공유해주실 것이 있으면 자유롭게 작성해 주세요.)
- post method 에서 Order, OrderList 두 모델을 만지는 부분이 있는데 Transaction 생성을 해주어야 할까요?
OrderList를 만들다 오류가 나더라도 장바구니는 만들어져도 상관없기 때문에 Transaction 처리를 안해주어도 크게 문제는 없을 듯 하긴 합니다! 그래도 해주는게 좋을까요?
<img width="569" alt="스크린샷 2021-06-11 오후 2 25 32" src="https://user-images.githubusercontent.com/55984573/121635245-e3d74680-cac0-11eb-80ba-f3a2a3914cf9.png">


## 체크 리스트 (아래 사항들이 전부 체크되어야만 merge가 됩니다!)
- [x] 필요한 test들을 완료하였고 기능이 제대로 실행되는지 확인 하였습니다.
- [x] Wecode의 코드 스타일 가이드에 맞추어 코드를 작성 하였습니다.
- [x] 제가 의도한 파일들과 수정 사항들만 커밋이 된 것을 확인 하였습니다.
- [x] 본 수정 사항들을 팀원들과 사전에 상의하였고 팀원들 모두 해당 PR에 대하여 알고 있습니다.

hyo-jin-jeong committed:
Add: 장바구니 추가, 삭제 기능 구현
- BasketView 생성
- post, get method 생성
- 예외 처리

hyo-jin-jeong committed:
Merge: main브랜치와 병합 후 충돌 해결

hyo-jin-jeong committed:
Remove: 불필요한 코드, 파일 제거

hyo-jin-jeong committed:
Add: 장바구니 중복 상품 검증 코드 작성

hyo-jin-jeong committed:
Modify: MultipleObjectsReturned status 수정

Kang-kyunghun commented on orders/views.py:
```diff
@@ -1,3 +1,68 @@
+            order_status = OrderStatus.objects.get(status='BASKET')
+            
+            order, created = Order.objects.get_or_create(
+                user=user, order_status=order_status,
```
```suggestion
                user = user, 
                order_status_id = OrderStatus.BASKET ,
```

hyo-jin-jeong replied:
수정 완료하였습니다!

hyo-jin-jeong committed:
Refactor: 멘토님 의견 반영
- endpoint 변경
- get() method filter()로 변경
- ObjectDoesNotExist 삭제
- OrderList model OrderItem으로 이름 변경

hyo-jin-jeong committed:
Modify: OrderStatus Model BASkET값 변경

hyo-jin-jeong committed:
Modify: Order view status code 수정

Kang-kyunghun commented on orders/views.py:
```diff
@@ -1,3 +1,66 @@
+            order_item = OrderItem.objects.get(id=order_item)
+
+            if order_item.order.user != request.user:
+                return JsonResponse({'message':'INVALID_ORDER_ITEMS'})
```
status code 추가해 주세요.

hyo-jin-jeong replied:
네! 수정했습니다

Kang-kyunghun commented:
질문 하신 Transaction 처리 부분은 저도 장바구니는 생성 되어도 무방하다고 생각합니다. 그래서 이 부분에서는 따로 transaction을 안 해도 될 것 같습니다.

hyo-jin-jeong committed:
Modify: status code 추가

Kang-kyunghun commented on orders/views.py:
```diff
@@ -1,3 +1,66 @@
+
+        except ObjectDoesNotExist:
+            return JsonResponse({'message':'DOES_NOT_EXIST'}, status=400)
+
+        
```
공백 지워주세요.

hyo-jin-jeong replied:
수정했습니다!

Kang-kyunghun commented on orders/views.py:
```diff
@@ -1,3 +1,66 @@
+        except MultipleObjectsReturned:
+            return JsonResponse({'message':'MULTIPLE_OBJECT_RETURNED'}, status=400)
+
+        except ObjectDoesNotExist:
+            return JsonResponse({'message':'DOES_NOT_EXIST'}, status=400)
```
except 말고 exists로 한 번 바꿔 보시겠어요?

hyo-jin-jeong replied:
수정했습니다!

Kang-kyunghun commented on orders/views.py:
```diff
@@ -1,3 +1,66 @@
+
+            return JsonResponse({'message':'SUCCESS'}, status=204)
+
+        except MultipleObjectsReturned:
+            return JsonResponse({'message':'MULTIPLE_OBJECT_RETURNED'}, status=400)
```
pk를 가지고 get을 하기 때문에 없어도 되는 예외처리로 보입니다.

hyo-jin-jeong replied:
수정했습니다!

hyo-jin-jeong committed:
Refactor: 불필요한 코드 제거

hyo-jin-jeong committed:
Refactor: 불필요한 코드 제거

hyo-jin-jeong committed:
Modify: token name 변경 -> Authorization

Kang-kyunghun commented:
LGTM~~!!
